mathesar/views/api.py

```diff
@@ -350,3 +359,52 @@ def create(self, request):
         serializer = DataFileSerializer(data=request.data, context={'request': request})
         serializer.is_valid(raise_exception=True)
         return create_datafile(request, serializer.validated_data)
+
+
+class ConstraintViewSet(viewsets.GenericViewSet, ListModelMixin, RetrieveModelMixin):
+    serializer_class = ConstraintSerializer
+    pagination_class = DefaultLimitOffsetPagination
+
+    def get_queryset(self):
+        return Constraint.objects.filter(table__id=self.kwargs['table_pk']).order_by('-created_at')
+
+    def create(self, request, table_pk=None):
+        table = get_table_or_404(table_pk)
+        serializer = ConstraintSerializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        # If we don't do this, the request.data QueryDict will only return the last column's name
+        # if there are multiple columns.
+        if type(request.data) != dict:
+            data = request.data.dict()
+            data['columns'] = request.data.getlist('columns')
+        try:
+            name = data['name'] if 'name' in data else None
```

Comment:
I don't think `data` is defined here.

Reply:
Ugh, sorry. It's fixed now, I verified through the HTML interface.